sql/updates/pre-global.sql

```diff
@@ -1,4 +1,3 @@
---Disable the event trigger during updates. Two reasons:
----1- Prevents the old extension .so being loaded during upgrade to process the event trigger.
----2- Probably the right thing to do anyway since you don't necessarly know which version of the trigger will be fired during upgrade.
-ALTER EVENT TRIGGER timescaledb_ddl_command_end DISABLE;
+--need to keep this legacy functions around for 0.9 since it's still used in 0.8, delete after 0.9
+CREATE OR REPLACE FUNCTION _timescaledb_internal.ddl_command_end() RETURNS event_trigger
```

Comment:
I am not sure I understand the behavior of this `pre-global.sql`. It seems it is only used to prepend stuff for the adjacent version you are upgrading from (e.g., 0.8.0--0.9.0), but not for prior versions. Instead you are modifying those upgrade scripts manually. This seems to be necessary also going forward since a function you are removing could have been added in an intermediate (not adjacent) version.

Since we are getting rid of this function, the "right" thing to do would be to DROP this the first thing we do in the upgrade scripts for all versions we upgrade from that have the function. Then we wouldn't need to keep the function around int `compat.c`. This actually seems to be the reason why we need `compat.c` in general. 

In fact, the way we generate "jump" update scripts by concatenating all intermediate scripts seems broken or, at least somewhat naive. Let's say we are upgrading from `0.6.0` to `0.9.0` but a function `foo_func` was added in `0.7.0` but then removed in `0.9.0`. The right thing to do is for the jump script to never define that function, thus not requiring its presence in the `0.9.0` so. What we have now is that we first require in the header that we redefine that function to point to 0.9.0, then redefine it again in 0.7.0, followed by a removal in 0.9.0 but still having to keep the C code in `compat.c`.